scripts/python_scripts/telegram_notifications.py

```diff
@@ -1,6 +1,36 @@
 import os
+import asyncio
 import telegram
 
-async def send_telegram_message(message):
-    bot = telegram.Bot(token=os.getenv("TELEGRAM_BOT_TOKEN"))
-    await bot.send_message(chat_id=os.getenv("TELEGRAM_CHAT_ID"), text=message, parse_mode="MarkdownV2")
+from subquery_api import SubQueryProject, DeploymentInstance
+
+
+class TelegramNotifications():
+    __instance = None
+    telegram_message = "⚠️ SubQuery projects error ⚠️"
+    need_to_send_notification = False
+
+    def __init__(self) -> None:
+        if not hasattr(self, 'initialized'):
+            self.initialized = True
+            self.token = os.getenv("TELEGRAM_BOT_TOKEN")
+            self.chat_id = os.getenv("TELEGRAM_CHAT_ID")
+
+    def __new__(cls):
+        if cls.__instance is None:
+            cls.__instance = super().__new__(cls)
+        return cls.__instance
```

Comment:
Why do we need those manipulations with cls.__instance and initialized flag? Is it an implementation of singleton? If so, we might consider using [metaclass](https://stackoverflow.com/questions/6760685/creating-a-singleton-in-python) instead

Reply:
yes, it is a singleton, I will do it with metaclass, thanks

Reply:
changed implementation in order to use metaclass singleton implementation